components/pages/blog/list.js

```diff
@@ -55,33 +60,41 @@ const Blog = () => {
             console.error('Error deleting post:', error);
         }
     };
+
     const renderAddPostButton = () => {
         if (authorized) {
             return (
-                <Button variant="outlined" color="secondary" onClick={() => openEditModalHandler({})}>
+                <Button variant="outlined" color="secondary" onClick={openAddModalHandler}>
                     Add Post
                 </Button>
             );
         }
         return null;
     };
+
     return (
         <Container>
             <Title>Welcome to My Blog</Title>
             {renderAddPostButton()}
             <PostListComponent
                 posts={posts}
                 isButtons={authorized}
-                openEditModalHandler={openEditModalHandler}
-                deletePost={handleDeletePost}
-            />
-            <PostModal
-                isModalOpen={isModalOpen}
-                closeModalHandler={closeModalHandler}
-                editedPost={editedPost}
                 handleEditPostAsync={handleEditPostAsync}
-                handleAddPost={handleAddPost}
+                deletePost={handleDeletePost}
+                openEditModalHandler={openEditModalHandler}
             />
+                <AddPostModal
+                    isModalOpen={isAddModalOpen}
+                    closeModalHandler={closeModalHandler}
+                    editedPost={{}}
+                    handleAddPost={handleAddPost}
+                />
+                <EditPostModal
+                    isModalOpen={isEditModalOpen}
+                    closeModalHandler={closeModalHandler}
+                    editedPost={editedPost}
+                    handleEditPostAsync={handleEditPostAsync}
+                />
```

Comment:
You can use the same Modal and Form for both creation and update. You just need to change the props a bit:
```
<EditPostModal
    title="Create post"
    submitButtonText="Add post"
    isModalOpen={isAddModalOpen}
    closeModalHandler={closeAddModalHandler}
    handleSubmit={handleAddPost}
/>
<EditPostModal
    title="Edit post"
    submitButtonText="Save changes"
    isModalOpen={!!postToEdit}
    closeModalHandler={closeEditModalHandler}
    handleSubmit={handleEditPost}
    initialData={postToEdit}
/>
```